EventsExpress.Core/Services/UserService.cs

```diff
@@ -239,6 +245,25 @@ public async Task EditGender(Gender gender)
             await Context.SaveChangesAsync();
         }
 
+        public async Task EditLocation(LocationDto location)
+        {
+            var user = CurrentUser();
+            if (user.LocationId == null)
+            {
+                var locationId = await _locationService.AddLocationToUser(location);
+                user.LocationId = locationId;
+            }
+            else
+            {
+                location.Id = (Guid)user.LocationId;
+            }
+
+            user.Location = Mapper.Map<LocationDto, Location>(location);
```

Comment:
I think we need to review and discuss this. It seems to me there is a possible reason for data inconsistency.